packages/perseus/src/util/parse-perseus-json/perseus-parsers/interaction-widget.ts

```diff
@@ -0,0 +1,186 @@
+    key: string,
+    options: object({
+        startX: string,
+        startY: string,
+        startSubscript: number,
+        endX: string,
+        endY: string,
+        endSubscript: number,
+        constraint: string,
+        snap: number,
+        constraintFn: string,
+        constraintXMin: string,
+        constraintXMax: string,
+        constraintYMin: string,
+        constraintYMax: string,
+    }),
+});
+
+type MovablePointElement = Extract<
+    PerseusInteractionElement,
+    {type: "movable-point"}
+>;
+const parseMovablePointElement: Parser<MovablePointElement> = object({
+    type: constant("movable-point"),
+    key: string,
+    options: object({
+        startX: string,
+        startY: string,
+        varSubscript: number,
+        constraint: string,
+        snap: number,
+        constraintFn: string,
+        constraintXMin: string,
+        constraintXMax: string,
+        constraintYMin: string,
+        constraintYMax: string,
+    }),
+});
+
+type ParametricElement = Extract<
+    PerseusInteractionElement,
+    {type: "parametric"}
+>;
+const parseParametricElement: Parser<ParametricElement> = object({
+    type: constant("parametric"),
+    key: string,
+    options: object({
+        x: string,
+        y: string,
+        rangeMin: string,
+        rangeMax: string,
+        color: string,
+        strokeDasharray: string,
+        strokeWidth: number,
+    }),
+});
+
+type PointElement = Extract<PerseusInteractionElement, {type: "point"}>;
+const parsePointElement: Parser<PointElement> = object({
+    type: constant("point"),
+    key: string,
+    options: object({
+        color: string,
+        coordX: string,
+        coordY: string,
+    }),
+});
+
+type RectangleElement = Extract<PerseusInteractionElement, {type: "rectangle"}>;
+const parseRectangleElement: Parser<RectangleElement> = object({
+    type: constant("rectangle"),
+    key: string,
+    options: object({
+        color: string,
+        coordX: string,
+        coordY: string,
+        width: string,
+        height: string,
+    }),
+});
+
+export const parseInteractionWidget: Parser<InteractionWidget> = parseWidget(
+    constant("interaction"),
+    object({
+        static: boolean,
+        graph: object({
+            editableSettings: optional(array(enumeration("canvas", "graph"))),
+            box: pairOfNumbers,
+            labels: array(string),
+            range: pair(pairOfNumbers, pairOfNumbers),
+            gridStep: pairOfNumbers,
+            markings: enumeration("graph", "grid", "none"),
+            snapStep: optional(pairOfNumbers),
+            valid: optional(union(boolean).or(string).parser),
+            backgroundImage: optional(parsePerseusImageBackground),
+            showProtractor: optional(boolean),
+            showRuler: optional(boolean),
+            rulerLabel: optional(string),
+            rulerTicks: optional(number),
+            tickStep: pairOfNumbers,
+        }),
```

Comment:
I want to avoid coupling different widgets to each other. Their options will likely diverge as we add features to Interactive Graph and deprecate the others.